tsdb/block.go

```diff
@@ -158,6 +158,9 @@ type BlockMeta struct {
 
 	// Version of the index format.
 	Version int `json:"version"`
+
+	// OutOfOrder is true if the block was directly created from out-of-order samples.
+	OutOfOrder bool `json:"out_of_order"`
```

Comment:
It is actually required and not for debugging. After a restart we truncate the head block (the in-order part) based on the last block present on disk. But we cannot consider the out of order blocks for head truncation because the out of order blocks can be in the future by 2h and can drop the in-order data in the head creating gaps (one of the production incidents we faced :|). So we need this to distinguish the blocks.

Reply:
Can you give an example of how it will look with enum?

Reply:
I like the idea of hints